### ISSUE TITLE
Application objects in Access generate false positive "Variable {x} is never used" inspection

### ISSUE THREAD

**Issue by daFreeMan** [rubberduck-vba/Rubberduck#708]
In Access: 

```
  Dim XLObj As Excel.Application
  Set XLObj = New Excel.Application
  XLObj.Calculation = xlCalculationManual
```

Generates a `Variable XLObj is never used` item in the _Code Inspections_

![rd never used 2](https://cloud.githubusercontent.com/assets/11889733/8624281/b859e6c2-2704-11e5-934b-ea5f60f39dad.PNG)

**Comment by retailcoder** [comment]
The resolver is working as expected - the type `Excel.Application` is unresolvable ("Excel" not having a `DeclarationType.Project` built-in declaration - see [VbaStandardLib.cs](https://github.com/rubberduck-vba/Rubberduck/blob/32f350d6ef0cf072db126a5400227c589d983708/Rubberduck.Parsing/Symbols/VbaStandardLib.cs#L34)) - we *could* (and probably *should*) add these `Declaration` objects though, and "model" the (gulp) *entire* Office object model, so that the resolver knows how to deal with those.

The problem is that to correctly resolve member calls, the resolver isn't making any assumption - if it can't resolve the *type* of `XLObj` (which it can't), then it stops trying to resolve the rest of the member call chain.

Clearly this is a bug in the resolver: even if the type of `XLObj` isn't known, a member call to `XLObj` *should* be picked up as a reference *regardless* of whether or not the actual member call chain is completely resolvable - in this case the `Calculation` member assignment.

I need to write a *few* (cough) unit tests before making any changes to that logic though.

**Comment by retailcoder** [comment]
This isn't a resolver issue - references to the `XLObj` variable are correctly resolved, at least in this repro:

![image](https://cloud.githubusercontent.com/assets/5751684/8625808/fa34b64e-270e-11e5-9e97-7661cfb7a1e2.png)

But the inspection doesn't agree: the bug is with the inspection.

**Comment by daFreeMan** [comment]
Update:

I just realized that I've defined `XLObj` in two different places:

1. It is a `Module` **global** in one module - references within this module are correctly identified, and I do _not_ get the `never used` inspection triggered
1. It is a `Sub` **local** in another module - this is the one that caused the false positive originally reported.

Both instances are `Dim XLObj as Excel.Application`
